src/gui/annotations/qgsannotationitemguiregistry.cpp

```diff
@@ -133,9 +133,15 @@ bool QgsAnnotationItemGuiRegistry::addItemGroup( const QgsAnnotationItemGuiGroup
   return true;
 }
 
-const QgsAnnotationItemGuiGroup &QgsAnnotationItemGuiRegistry::itemGroup( const QString &id )
+const QgsAnnotationItemGuiGroup &QgsAnnotationItemGuiRegistry::itemGroup( const QString &id ) const
 {
-  return mItemGroups[id];
```

Comment:
Isn't this a different meaning now? Previously if you asked for a group with a non-existing id, then one (a valid one) would be created and returned. Now if you ask for a group with a non-existing id you'll always get the same invalid group returned.